packages/playwright/src/matchers/expect.ts

```diff
@@ -338,6 +340,7 @@ class ExpectMetaInfoProxyHandler implements ProxyHandler<any> {
       };
 
       const finalizer = () => {
+        setCurrentStep(undefined);
```

Comment:
We can't use this pattern as steps assume nesting and this would be clearing parent upon exiting child. We need to rely on step hierarchy as managed by test step.